src/vec2.rs

```diff
@@ -314,6 +320,32 @@ impl Vec2 {
             y: self.y / divisor,
         }
     }
+
+    /// Turn by 90 degrees.
+    ///
+    /// The rotation is clockwise in a Y-down coordinate system. The following relations hold:
+    ///
+    /// `u.dot(v) = u.cross(v.turn_90())`
+    ///
+    /// `u.cross(v) = u.turn_90().dot(v)`
+    #[inline]
+    pub fn turn_90(self) -> Vec2 {
+        Vec2::new(-self.y, self.x)
+    }
+
+    /// Combine two vectors interpreted as rotation and scaling.
+    ///
+    /// Interpret both vectors as a rotation and a scale, and combine
+    /// their effects. This operation is equivalent to multiplication
+    /// when the vectors are interpreted as complex numbers. It is
+    /// commutative.
```

Comment:
It might be worth it to state explicitly that the angles are added, and the magnitudes are multiplied.

Maybe:

```suggestion
    /// Combine two vectors interpreted as rotation and scaling.
    ///
    /// Interpret both vectors as a rotation and a scale, and combine
    /// their effects by adding the angles and multiplying the magnitudes.
    /// This operation is equivalent to multiplication when the vectors
    /// are interpreted as complex numbers. It is commutative.
```